connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/WorkerRebalanceListener.java

```diff
@@ -36,4 +37,15 @@ public interface WorkerRebalanceListener {
      * or tasks might refer to all or some of the connectors and tasks running on the worker.
      */
     void onRevoked(String leader, Collection<String> connectors, Collection<ConnectorTaskId> tasks);
+
+
+    /**
+     * Invoked when a worker experiences a poll timeout expiry. Invoking this method allows getting
+     * the stage which was currently being executed when the poll timeout happened. The default implementation
+     * returns null
+     * @return The current stage being executed. Could be null
+     */
+    default Stage onPollTimeoutExpiry() {
```

Comment:
This is an internal interface, unless this default method actually makes sense on it's own I wouldn't add it.

Reply:
I added the default bit to avoid changing the tests. Turns out it should be ok to modify them so I removed the default bit.